packages/ove-asset-manager/src/OVE.Service.AssetManager/Controllers/OVEAssetModelController.cs

```diff
@@ -0,0 +1,517 @@
+        /// </summary>
+        /// <param name="id">id of the asset</param>
+        /// <returns>url of the asset</returns>
+        [HttpGet]
+        [Route("/OVEAssetModelController/GetAssetURLbyId/")]
+        public  async Task<ActionResult<string>> GetAssetUrl(string id) {
+            return await GetAssetUrl(m => m.Id==id);
+        }
+
+        #endregion 
+
+        private async Task<ActionResult<string>> GetAssetUrl(Expression<Func<OVEAssetModel, bool>> expression) {
+            var assetModel = await _context.AssetModels.Where(expression).OrderByDescending(m=> m.LastModified).FirstOrDefaultAsync();
+
+            if (assetModel == null) {
+                return NotFound();
+            }
+
+            return _fileOperations.ResolveFileUrl(assetModel);
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Return the index page
+        /// </summary>
+        /// <returns>index page</returns>
+        [HttpGet]
+        [Route("/OVEAssetModelController/")]
+        public async Task<IActionResult> Index() {
+            return View(await _context.AssetModels.ToListAsync());
+        }
+
+        /// <summary>
+        /// Get Details of the asset by guid 
+        /// assetModels/Details/5
+        /// </summary>
+        /// <param name="id">guid of the asset </param>
+        /// <returns>details of the asset</returns>
+        [HttpGet]
+        [Route("/OVEAssetModelController/Details/{id}.{format?}")]
+        public async Task<ActionResult<OVEAssetModel>> Details(string id) {
+            if (id == null) {
+                return NotFound();
+            }
+
+            var assetModel = await _context.AssetModels
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (assetModel == null) {
+                return NotFound();
+            }
+
+            return this.FormatOrView(assetModel);
+        }
+
+        /// <summary>
+        /// Returns the create page
+        /// GET: assetModels/Create
+        /// </summary>
+        /// <returns>creation gui page</returns>
+        [HttpGet]
+        [Route("/OVEAssetModelController/Create")]
+        public IActionResult Create() {
+            return View();
+        }
+
+        /// <summary>
+        /// Post a new asset
+        /// Simplest Post is form-data with upload (file) with Project,Name and Service
+        ///  </summary>
+        /// <param name="oveAssetModel">asset model (Project, Name, Description)</param>
+        /// <param name="upload">the asset file to upload</param>
+        /// <returns></returns>
+        [HttpPost]
+        [DisableRequestSizeLimit]
+        [Route("/OVEAssetModelController/Create/{format?}")]
+        public async Task<ActionResult<OVEAssetModel>> Create(
+            [Bind("Project,Name,Description,Service,AssetMeta")] OVEAssetModel oveAssetModel,
+            [FromForm] IFormFile upload) {
+
+            // check if we have a file
+            if (upload == null || upload.Length <= 0) {

```

Comment:
@davidbirchwork, can you demonstrate how I can test the case of uploading a zero-byte file and getting this error? I have tried with an empty file (I cannot e-mail or attach since it is empty), and it doesn't show me any error.

<img width="460" alt="image" src="https://user-images.githubusercontent.com/3415383/46860979-6f2c4580-ce09-11e8-8040-e0b23e9a4fc1.png">